helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -97,11 +97,12 @@ public ParticipantManager(HelixManager manager, HelixZkClient zkclient, int sess
   }
 
   /**
-   * Handle new session for a participang.
+   * Handle new session for a participant.
    * @throws Exception
    */
   public void handleNewSession() throws Exception {
-    joinCluster();
+
+    autoRegistration();
```

Comment:
Let's not rush for auto registration logic here. We can split it into two PRs. For this PR, let's only have the interface and class implementation. Move the registration logic to separate PR.